src/types/type_param.rs

```diff
@@ -39,29 +34,21 @@ pub enum TypeParam {
 pub enum TypeArg {
     /// Where the (Type/Op)Def declares that an argument is a [TypeParam::Type]
     Type(SimpleType),
-    /// Where the (Type/Op)Def declares that an argument is a [TypeParam::ClassicType],
-    /// it'll get one of these (rather than embedding inside a Type)
-    ClassicType(ClassicType),
-    /// Where the (Type/Op)Def declares that an argument is a [TypeParam::HashableType],
-    /// this is the value.
-    HashableType(HashableType),
-    /// Where the (Type/Op)Def declares a [TypeParam::Value] of type [HashableType::Int], a constant value thereof
-    Int(HugrIntValueStore),
-    /// Where the (Type/Op)Def declares a [TypeParam::Value] of type [HashableType::String], here it is
-    String(String),
-    /// Where the (Type/Op)Def declares a [TypeParam::List]`<T>` - all elements will implicitly
-    /// be of the same variety of TypeArg, i.e. `T`s.
+    /// 64-bit unsigned integer
+    Usize(u64),
+    /// An opaque value, stored as serialized blob.
+    Opaque(CustomSerialized),
+    /// List of arguments
     List(Vec<TypeArg>),
-    /// Where the TypeDef declares a [TypeParam::Value] of [Container::Opaque]
-    CustomValue(serde_yaml::Value),
+    /// Tuple of arguments
```

Comment:
Note that given we are only checkable not synthesizing here, we can combine TypeArg::List and TypeArg::Tuple into a single TypeArg (e.g. Sequence). We do this for constant values and I think there's an argument for this.